### PR TITLE
Changing return code from -1 to 1

### LINKED ISSUE #2707: Don't return -1 exit code from managed code
Follow up to https://github.com/aspnet/dnx/pull/2691 but in managed code - on non-windows valid exit codes are 0-255. However, in several places we return `-1` which is then converted to 255 which is wrong. We should return 1.
(Especially review all `return -1` and `Task.FromResult(-1)` occurrences)

### PR DESCRIPTION
**Pull request by moozzyk**
Fixes #2707

### CODE REVIEW

**Comment by muratg** [comment]
:shipit:

**Comment by troydai** [comment]
Any tests?

**Comment by moozzyk** [comment]
@troydai - existing tests are all passing and some of them are checking that the exit code is not 0. I can probably add tests for some conditions (e.g. not sure about this: https://github.com/aspnet/dnx/pull/2865/files#diff-0b1faa794a3c5fb04d329d96e22f8391L55) if you convince me that they add value.

**Comment by JunTaoLuo** [comment]
For the current tests that assert exit code is not 0, can we make it more specific and assert that they are equal to 1 where appropriate? I was trying to do this previously but we had a mix of -1 and 1 then.

**Comment by troydai** [comment]
Test non-zero. That makes sense. I was wondering why there is no test broken. Only thing I'd like to have covered in tests is that instead of testing non-zero how about testing `>0` so as to cover the essence of this change.

**Comment by moozzyk** [comment]
@JunTaoLuo - I can try that.

**Comment by moozzyk** [comment]
Testing `> 0` is not much better than `!= 0` I think. But I will try what @JunTaoLuo proposed and see how far I can get.

**Comment by JunTaoLuo** [comment]
Also, should we just go ahead and make `dnu` consistent as well? 
E.g. https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Tooling/ConsoleCommands/BuildConsoleCommand.cs#L49
Same with `PackConsoleCommand` and `PublishConsoleCommand`

**Comment by moozzyk** [comment]
I checked that - this value is not being used anywhere let alone returned as exit code so I left it as is.

**Comment by JunTaoLuo** [comment]
I agree we currently don't use them anywhere but I think they are being returned as exit codes:

image

**Comment by moozzyk** [comment]
@JunTaoLuo - now I see this. I missed that they are returned from the OnExecute and not from the `Register` function. Will fix that because as you said - they are being returned as exit code.

**Comment by moozzyk** [comment]
@JunTaoLuo @troydai - Updated the PR:
- all the tests that used `NotEqual(0, exitCode)` are checking against the expected exit code (1 or 2)
- all the commands now return 1 in case of errors (good catch @JunTaoLuo - I missed the `}` and therefore thought that we never checked/used the value)

File: test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPackTests.Old.cs [hunk 1]
```diff
@@ -475,7 +475,7 @@ public void DnuPack_ShowUnresolvedDependencyWhenBuildFails(string flavor, string
                     $"{projectJsonPath} -s {emptyLocalFeed}",
                     out stdOut,
                     out stdErr);
-                Assert.NotEqual(0, exitCode);
```
Comment (JunTaoLuo) [inline test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPackTests.Old.cs]: Missed the `Assert.NotEqual(0, exitCode);` on line 487

File: test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPackTests.Old.cs [hunk 2]
```diff
@@ -475,7 +475,7 @@ public void DnuPack_ShowUnresolvedDependencyWhenBuildFails(string flavor, string
...
+                Assert.Equal(1, exitCode);
```
Comment (JunTaoLuo) [inline test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPackTests.Old.cs]: Missed the `Assert.NotEqual(0, exitCode);` on line 487

Comment (moozzyk) [inline test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPackTests.Old.cs]: Fixed.

**Comment by davidfowl** [comment]
Travis?

**Comment by muratg** [comment]
Travis failure seems like a timeout, no?

**Comment by davidfowl** [comment]
Then force a re-run

**Comment by troydai** [comment]
I kicked off a re-run. You're welcome.

**Comment by troydai** [comment]
Looks good to me. :shipit: if Travis passes.

**Comment by muratg** [comment]
One of the runs seem passed.  Not sure why it' s not reflected here.

**Comment by JunTaoLuo** [comment]
Not sure what has happened with travis but the Linux run seem to error due to timeout very frequently. It usually takes 2-3 tries to get a success.

**Comment by davidfowl** [comment]
@JunTaoLuo I think we have an actual bug somewhere.

**Comment by davidfowl** [comment]
:shipit:

**Comment by moozzyk** [comment]
@muratg @troydai  - thanks for following up and kicking off the travis build!

**Pull request merged**